sycl/test-e2e/ESIMD/unified_memory_api/Inputs/atomic_update_slm.hpp

```diff
@@ -116,14 +113,14 @@ bool test_slm(queue q) {
     auto e = q.submit([&](handler &cgh) {
       cgh.parallel_for(rng, [=](id<1> ii) SYCL_ESIMD_KERNEL {
         int i = ii;
-        slm_init<32768>();
+        constexpr uint32_t SLMSize = N * (stride + 1) * sizeof(T);
+        slm_init<SLMSize>();
+
         simd<uint32_t, N> offsets(start_ind * sizeof(T), stride * sizeof(T));
         simd<T, size> data;
         data.copy_from(arr);
 
-        simd<uint32_t, size> slm_offsets(0, sizeof(T));
-        // TODO: replace this API with unified version.
-        lsc_slm_scatter(slm_offsets, data);
+        slm_block_store(0, data);
```

Comment:
It is non atomic operation and each of threads does the same work in parallel, thus potentially might be problems with it.
It would be safer if only 0-th thread do init and read-for-verification. I.e. here:
```suggestion
        if (ndid.get_local_id(0) == 0)
          slm_block_store(0, data);
```
Also, it is better to use SLM with ND-Range parallel_for.
For example, it is the requirement for local_accessors.

Reply:
Good, point, thanks!